File: BE_QuanLyTro/src/main/java/ut/edu/be_quanlytro/Dto/Request/ContractCreateRequest.java
package ut.edu.be_quanlytro.Dto.Request;

import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

@Getter @Setter
public class ContractCreateRequest {
    private UUID roomId;
    private LocalDate startDate;
    private LocalDate endDate;
    private BigDecimal depositAmount;
    private String tenantPhone;
    private UUID depositId;
}

File: Flutter_QuanLyTro/lib/data/models/request/contract_create_request.dart
import 'dart:convert';

class ContractCreateRequest {
  final String roomId;
  final String? depositId;
  final String tenantPhone;
  final String startDate;
  final String endDate;
  final double depositAmount;

  ContractCreateRequest({
    required this.roomId,
    this.depositId,
    required this.tenantPhone,
    required this.startDate,
    required this.endDate,
    required this.depositAmount,
  });

  Map<String, dynamic> toMap() {
    return {
      'roomId': roomId,
      if (depositId != null) 'depositId': depositId,
      'tenantPhone': tenantPhone,
      'startDate': startDate,
      'endDate': endDate,
      'depositAmount': depositAmount,
    };
  }

  String toJson() => json.encode(toMap());
}